### PR TITLE
fixed bug in drop_features

### PR DESCRIPTION
**Pull request by luismavs**
Hi,

I found a bug in the DropFeatures transformer, module drop_features.py. The check (number of columns) in the .fit() method may fail unintentionally if features_to_drop is a string.

```
from feature_engine.selection import DropFeatures
import pandas as pd

df = pd.DataFrame({'aa':[1,2], 'bb': [3,4]})
transformer = DropFeatures(features_to_drop="aa")
X = transformer.fit_transform(df)

```

To fix this, I´ve applied the conversion of features_to_drop using the _find_all_variables method.

### CODE REVIEW

File: feature_engine/selection/drop_features.py [hunk 1]
```diff
@@ -81,10 +81,11 @@ def fit(self, X: pd.DataFrame, y: pd.Series = None):
         # present in the df.
         X[self.features_to_drop]
 
-        self.features_to_drop_ = self.features_to_drop
+        self.features_to_drop_ = _find_all_variables(X, variables=self.features_to_drop)
+        # self.features_to_drop_ = self.features_to_drop
```
Comment (solegalli) [inline feature_engine/selection/drop_features.py:85]: could you please remove the commented line?

**Review by solegalli — COMMENTED** [review]
Hey @luismavs thank you so much! 

This is very helpful.

Could I ask you a few small things?

First, we need to fix code style. Here the error:

```
feature_engine/selection/drop_features.py:8:89: E501 line too long (89 > 88 characters)
feature_engine/selection/drop_features.py:10:1: E302 expected 2 blank lines, found 1
```

In short, we need to shorten 1 line and add a blank line to the other.

And also, could you use the syntax that you have in your comment to create a new test?

So that we ensure that we do not incur accidentally in the same error if we modify the class at some point.

Thank you!!

File: feature_engine/selection/drop_features.py (hunk 1 above)
Comment (luismavs) [reply feature_engine/selection/drop_features.py]: Hi,

I´ve pushed a new commit with the formatting changes you asked for, and an expanded version of the test test_drop_1_variable(...) which demonstrates the fix ( len('Marks') = len(X.columns) which was triggering the bug).

**Comment by solegalli** [comment]
Hey @luismavs 

Thanks! There are still some codestyle errors.

Do you have installed a tool called [black](https://pypi.org/project/black/)?

If you do, you just need to execute `black feature-engine tests` and it will resolve it automatically (almost always).

To reproduce the errors locally, you need to install [flake](https://pypi.org/project/flake8/) and then execute `flake8 feature-engine tests`

Check the docs [here](https://feature-engine.readthedocs.io/en/latest/contribute/contribute_code.html#test-code-style)

These are the errors if you want to resolve manually:

```
feature_engine/selection/drop_features.py:86:56: W291 trailing whitespace
feature_engine/selection/drop_features.py:87:53: E128 continuation line under-indented for visual indent
feature_engine/selection/drop_features.py:112:1: W391 blank line at end of file
tests/test_selection/test_drop_features.py:38:1: E303 too many blank lines (3)
tests/test_selection/test_drop_features.py:98:1: W391 blank line at end of file
stylechecks: exit 1 (3.46 seconds) /home/circleci/project> flake8 feature_engine tests pid=454
  stylechecks: FAIL code 1 (6.24=setup[2.78]+cmd[3.46] seconds)
  evaluation failed :( (6.33 seconds)


Exited with code exit status 1

CircleCI received exit code 1

```

**Comment by luismavs** [comment]
Hey, code has now been properly formatted  with black, looks like good to go.

File: tests/test_selection/test_drop_features.py [hunk 2]
```diff
@@ -26,6 +26,13 @@ def test_drop_1_variable(df_vartypes):
     assert type(X) == pd.DataFrame
     pd.testing.assert_frame_equal(X, df)
 
+    # checking features_to_drop has been correctly cast to list of str internally
+    transformer = DropFeatures(features_to_drop='Marks')
```
Comment (solegalli) [inline tests/test_selection/test_drop_features.py:30]: Hi, this looks good, but it took me a while to understand what this test was about, and why it is different from that in line 8.

Could we, instead of making it part of the current test, have a different test, just for this issue, with a clear name of what is being tested, and also in a comment add a link to the issue or PR?

File: tests/test_selection/test_drop_features.py [hunk 3]
```diff
@@ -26,6 +26,13 @@ def test_drop_1_variable(df_vartypes):
...
+    X = transformer.fit_transform(df_vartypes)
+    assert X.shape == (4, 4)
+    assert transformer.features_to_drop == "Marks"
+    assert type(X) == pd.DataFrame
+
```
Comment (solegalli) [inline tests/test_selection/test_drop_features.py:35]: could we please add an assert to compare the result of the transformation with what we expect from it? see line 27

File: feature_engine/encoding/decision_tree.py [hunk 4]
```diff
@@ -199,7 +199,6 @@ def __init__(
         variables: Union[None, int, str, List[Union[str, int]]] = None,
         ignore_format: bool = False,
     ) -> None:
-
```
Comment (solegalli) [inline feature_engine/encoding/decision_tree.py:202]: this file was probably modified with black, but it is not really part of this PR. Could we try and remove it please?

File: feature_engine/selection/drop_features.py [hunk 5]
```diff
@@ -5,6 +5,9 @@
 from feature_engine.dataframe_checks import check_X
 from feature_engine.selection.base_selector import BaseSelector
 from feature_engine.tags import _return_tags
+from feature_engine._variable_handling.variable_type_selection import (
```
Comment (solegalli) [inline feature_engine/selection/drop_features.py:8]: Could we also run isort to order the imports?

`isort path/to/file`

or alternatively order the imports manually. The last one starts with _, so it should be at the top of the feature engine imports more or less.

**Review by solegalli — COMMENTED** [review]
Hi @luismavs 

This is looking really good. Thank you!

I've got a couple of cosmetic requests and a kind of important favour to ask for the test.

Would you be able to have a look?

Thank you!

We are almost there :)

File: feature_engine/selection/drop_features.py (hunk 5 above)
Comment (luismavs) [reply feature_engine/selection/drop_features.py:5]: alright, done.

**Comment by solegalli** [comment]
Thank you @luismavs !!

It's all done now.

That was a nasty bug you found :)
